presto-hive/src/main/java/io/prestosql/plugin/hive/GenericHiveRecordCursor.java

```diff
@@ -278,10 +278,15 @@ private void parseLongColumn(int column)
             nulls[column] = true;
         }
         else {
-            Object fieldValue = ((PrimitiveObjectInspector) fieldInspectors[column]).getPrimitiveJavaObject(fieldData);
-            checkState(fieldValue != null, "fieldValue should not be null");
-            longs[column] = getLongExpressedValue(fieldValue, hiveStorageTimeZone);
-            nulls[column] = false;
+            try {
+                Object fieldValue = getPrimitiveObjectValue(column, fieldData);
+                checkState(fieldValue != null, "fieldValue should not be null");
+                longs[column] = getLongExpressedValue(fieldValue, hiveStorageTimeZone);
+                nulls[column] = false;
+            }
+            catch (IllegalArgumentException e) {
+                throw new PrestoException(HIVE_BAD_DATA, String.format("Error parsing field value '%s' for field %s: %s", fieldData, column, e.getMessage()), e);
```

Comment:
Move `Object fieldValue = getPrimitiveObjectValue(column, fieldData);` before the try catch block.
Then you can catch all RTE here

Reply:
The same comment applies here (as discussed below) about including the original error message in the newly thrown exception message.